sched/test_hash_stress.py

```diff
@@ -1,30 +1,32 @@
 """
-Hash scheduler pins resources to specific servers and connections. Stress
-test. Can't track server connections here, but real HTTP servers and clients
-are used.
+Test for Hash scheduler under heavy load. Uri should be pinned to a single
+server connection. Primary server connection should get the full load.
+When the connection is down, other connection is to be loaded. One primary
+connection is back online it should again get all the load.
 """
 
 import sys
 from helpers import tempesta
 from testers import stress
 
 __author__ = 'Tempesta Technologies, Inc.'
-__copyright__ = 'Copyright (C) 2017 Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2017-2018 Tempesta Technologies, Inc.'
 __license__ = 'GPL2'
 
 
 class BindToServer(stress.StressTest):
-    """ Hash scheduler binds URIs to specific connections, so only one server
-    must pull all the load if we try to get the same resource over and over
-    again.
+    """ Send requests with the same URI only one connection should be loaded,
+    but a few other connections can get a little bit of the load.
```

Comment:
It seems that in fact we are verifying loading for one server - not for one connection (according for current test configuration, there are 32 connections for each server).

Reply:
Yes, but we just can't get per-connection statistics from the backend. Backend doesn't see any difference between clients and Tempesta, and can't distinguish one connection from another. I could enable access log on Nginx, which is currently disabled for performance reasons, and analyse it. Frankly, this is too complicated approach for the tests.

On the other hand, there is a bunch of pure functional tests and they check that each resource is "pinned" to a single connection, not server. Thus very strict checks are not required in this particular test. Tests in this file are to confirm extra conditions:
- load balancing works as expected under pretty good load;
- load balancing works as expected if server closes connection time to time. 

Since this point wasn't clear, I'll extend description of this test.

Reply:
Test description was extended.